utils/runtime.go

```diff
@@ -0,0 +1,110 @@
+package utils
+
+import (
+	"context"
+	"fmt"
+	"github.com/go-redis/redis/v7"
+	sharedpb "github.com/tron-us/go-btfs-common/protos/shared"
+	"github.com/tron-us/go-common/db/postgres"
+	dbenv "github.com/tron-us/go-common/env/db"
+	"github.com/tron-us/go-common/log"
+	"go.uber.org/zap"
+	"os"
+	"strconv"
+)
+
+const (
+	DBConnectionHealthy = "DB Connection Healthy"
+	DBWriteConnectionError = "Cannot connect to the write database!"
+	DBReadConnectionError  = "Cannot connect to the read database!"
+	DBEnvironmentError = "Cannot parse the database environment variable"
+	RedisEnvironmentError = "Cannot parse the redis database environment variable"
+	RedisConnectionError = "Cannot connect to the redis database!"
+)
+
+var (
+	REDISClient *redis.Client
+	RedisHost     = "127.0.0.1"
+	RedisPort     = 6379
+	RedisPassword = ""
+	RedisDb       = 0
+)
+
+func getEnv(key, fallback string) string {
+	value := os.Getenv(key)
+	if len(value) == 0 {
+		return fallback
+	}
+	return value
+}
+
+func ConnectRedis(url string, port int, password string, db int) *redis.Client {
+	var client = redis.NewClient(&redis.Options{
+		Addr:     url + ":" + strconv.Itoa(port),
+		Password: password, // no password set
+		DB:       db,       // use default DB
+	})
+	return client
+}
+
+func CheckRedisConnection(connection *redis.Client) error {
+	val, err := connection.Ping().Result()
+	if err != nil {
+		return err
+	}
+	if val != "PONG" {
+		return fmt.Errorf("Redis respond error %s %d ", RedisHost, RedisPort)
+	}
+	return err
+}
+
+func CheckRuntime(ctx context.Context, runtime *sharedpb.RuntimeInfoRequest) (*sharedpb.RuntimeInfoReport, error) {
+	// db runtime
+	report := new(sharedpb.RuntimeInfoReport)
+
+	// Check postgres dbWrite
+	report.DbStatusExtra = []byte(DBConnectionHealthy)
```

Comment:
Here if the service doesn't have database, the DBConnectionHealthy will look weird.
I would suggest check env first, if no database you can leave it as nil.

Reply:
ok